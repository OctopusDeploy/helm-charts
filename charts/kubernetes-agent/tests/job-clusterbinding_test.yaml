suite: "job permissions"
templates:
- templates/job-clusterbinding.yaml
tests:
- it: "is not created when role rules are set"
  set:
    jobServiceAccount:
      role:
        rules:
        - apiGroups:
          - '*'
          resources:
          - ["pods"]
          verbs:
          - ["get", "watch", "list"]
  asserts:
  - hasDocuments:
      count: 0

- it: "should match snapshot"
  asserts:
  - matchSnapshot: {}
