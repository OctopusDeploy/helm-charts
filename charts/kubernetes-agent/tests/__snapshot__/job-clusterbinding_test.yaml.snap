should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: octopus-agent-job-RELEASE-NAME-binding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: octopus-agent-job-RELEASE-NAME-role
    subjects:
      - kind: ServiceAccount
        name: octopus-agent-job
        namespace: NAMESPACE
